java/ql/src/Security/CWE/CWE-094/InsecureBeanValidation.ql

```diff
@@ -14,6 +14,45 @@ import semmle.code.java.dataflow.TaintTracking
 import semmle.code.java.dataflow.FlowSources
 import DataFlow::PathGraph
 
+/**
+ * A message interpolator Type that perform Expression Language (EL) evaluations
+ */
+class ELMessageInterpolatorType extends RefType {
+  ELMessageInterpolatorType() {
+    this
+        .getASourceSupertype*()
+        .hasQualifiedName("org.hibernate.validator.messageinterpolation",
+          ["ResourceBundleMessageInterpolator", "ValueFormatterMessageInterpolator"])
+  }
+}
+
+/**
+ * A method call that sets the application's default message interpolator to an interpolator type that is likely to be safe,
+ * because it does not process Java Expression Language expressions.
+ */
+class SetSafeMessageInterpolatorCall extends MethodAccess {
+  SetSafeMessageInterpolatorCall() {
+    exists(Method m, RefType t |
+      this.getMethod() = m and
+      m.getDeclaringType().getASourceSupertype*() = t and
+      (
+        t.hasQualifiedName("javax.validation", ["Configuration", "ValidatorContext"]) and
+        m.getName() = "messageInterpolator"
+        or
+        t
+            .hasQualifiedName("org.springframework.validation.beanvalidation",
+              ["CustomValidatorBean", "LocalValidatorFactoryBean"]) and
+        m.getName() = "setMessageInterpolator"
+      )
+    ) and
+    not this.getAnArgument().getType() instanceof ELMessageInterpolatorType
```

Comment:
I suggest we take this logic out of the characteristic predicate, and put it into a `predicate isSafe()` on this class. Document it with the QLDoc that you currently have on the class.